### PR TITLE
merge lease to EtcdClient and lease test

### PR DESCRIPTION
**Pull request by StupidHod**
With ISSUE #61 , I pull this merge request 
merge lease to EtcdClient
add testing for lease.
@xiang90 @AdoHe . Look forward your review.

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/EtcdLeaseImpl.java [hunk 1]
```diff
@@ -184,6 +185,24 @@ public synchronized void keepAlive(long leaseId, EtcdLeaseHandler etcdLeaseHandl
...
 
     /**
+     * cancel keep alive for lease in background
+     *
+     * @param leaseId id of lease to set handler
+     */
+    @Override
+    public synchronized void cancelKeepAlive(long leaseId) throws ExecutionException, InterruptedException {
+        if(!isKeepAliveServiceRunning()){
+            throw new IllegalStateException("Lease keep alive service not start yet");
```
Comment (urcadox) [inline src/main/java/com/coreos/jetcd/EtcdLeaseImpl.java]: not started* yet

File: src/main/java/com/coreos/jetcd/EtcdLeaseImpl.java [hunk 2]
```diff
@@ -184,6 +185,24 @@ public synchronized void keepAlive(long leaseId, EtcdLeaseHandler etcdLeaseHandl
...
+        }
+        if (this.keepAlives.containsKey(leaseId)) {
+            keepAlives.remove(leaseId);
+            revoke(leaseId).get();
+        }else{
+            throw new IllegalStateException("Lease doesn't registered in keep alive service");
```
Comment (urcadox) [inline src/main/java/com/coreos/jetcd/EtcdLeaseImpl.java]: => Lease is not registered in the keep alive service

File: src/main/java/com/coreos/jetcd/EtcdLeaseImpl.java [hunk 3]
```diff
@@ -172,9 +168,14 @@ public boolean isKeepAliveServiceRunning() {
      *
      * @param leaseId          id of lease to set handler
      * @param etcdLeaseHandler the handler for the lease, this value can be null
+     *
+     * @throws IllegalStateException this exception hints that the keep alive service wasn't started
      */
     @Override
     public synchronized void keepAlive(long leaseId, EtcdLeaseHandler etcdLeaseHandler) {
+        if(!isKeepAliveServiceRunning()){
+            throw new IllegalStateException("Lease keep alive service not start yet");
```
Comment (urcadox) [inline src/main/java/com/coreos/jetcd/EtcdLeaseImpl.java]: not started* yet

**Comment by xiang90** [comment]
@StupidHod Fix the comments from @urcadox, then lgtm

**Comment by StupidHod** [comment]
@xiang90 @urcadox , thanks for your comments. I have fixed the comments.

**Pull request merged**